manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -37,18 +37,37 @@ type TaskReaper struct {
 	// List of tasks collected for cleanup, which includes two kinds of tasks
 	// - serviceless orphaned tasks
 	// - tasks with desired state REMOVE that have already been shut down
-	cleanup  []string
-	stopChan chan struct{}
-	doneChan chan struct{}
+	cleanup []string
+	// Map of services marked for removal. These are services that have
+	// been marked for removal, and have tasks that are in the process of
+	// shutting down - specifically tasks with desired state REMOVE.
+	cleanupServices map[string]struct{}
+	stopChan        chan struct{}
+	doneChan        chan struct{}
 }
 
 // New creates a new TaskReaper.
 func New(store *store.MemoryStore) *TaskReaper {
 	return &TaskReaper{
-		store:    store,
-		dirty:    make(map[orchestrator.SlotTuple]struct{}),
-		stopChan: make(chan struct{}),
-		doneChan: make(chan struct{}),
+		store:           store,
+		dirty:           make(map[orchestrator.SlotTuple]struct{}),
+		stopChan:        make(chan struct{}),
+		doneChan:        make(chan struct{}),
+		cleanupServices: make(map[string]struct{}),
+	}
+}
+
+// addServiceForCleanup checks if the service is marked for removal, and if it is,
+// adds it to the map cleanupServices so that it can be considered for deletion later.
+// This is only done if the service doesn't already exist in cleanupServices.
+func (tr *TaskReaper) addServiceForCleanup(id string) {
```

Comment:
Is this function still used?

Reply:
Nope, will remove it. I must have forgotten.